container/ClearSky/src/main/java/io/egen/clearskyboot/services/implementations/WeatherServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+package io.egen.clearskyboot.services.implementations;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.egen.clearskyboot.entities.Reading;
+import io.egen.clearskyboot.exceptions.InternalServerError;
+import io.egen.clearskyboot.exceptions.NotFoundException;
+import io.egen.clearskyboot.repositories.WeatherRepository;
+import io.egen.clearskyboot.services.WeatherService;
+
+@Service
+@Transactional
+public class WeatherServiceImpl implements WeatherService {
+	
+	@Autowired
+	private WeatherRepository weatherRepository;
+
+	@Override
+	public List<Reading> findAll() {
+		return weatherRepository.findAll();
+	}
+
+	@Override
+	public Reading create(Reading reading) {
+		return weatherRepository.save(reading);
+	}
+
+	@Override
+	public List<String> findDistinctCities() {
+		return weatherRepository.findDistinctCities();
+	}
+
+	@Override
+	public Reading findLatestWeatherByCity(String city) {
+		Optional<Reading> optionalReading = Optional.ofNullable(weatherRepository.findTop1ByCityOrderByTimestampDesc(city));
+		if(optionalReading.isPresent()) return optionalReading.get();
+		throw new NotFoundException("No latest data found for city : " + city);
+	}
+
+	@Override
+	public JsonNode findLatestWeatherPropertyByCity(String city, String property) {
```

Comment:
Found a way to do this ! I've used the existing 'Reading' entity to have the above custom properties as well ! The catch is I've annotated these properties as @Transient (javax.persistence.Transient). This helped to create custom properties in existing entity but NOT persist these custom properties in database.